src/datafile.rs

```diff
@@ -133,12 +123,7 @@ impl DataFile {
     }
 
     pub fn sync(&mut self) -> ErrorResult<()> {
-        let res = self.file.sync_all();
-        if res.is_ok() {
-            return Ok(());
-        }
-
-        Err(Box::new(res.err().unwrap()))
+        self.file.sync_all().map_err(Into::into)
```

Comment:
we can just map the error into your `Box<dyn std::error::Error>` here. making this way simpler.